src/Lib/Element.php

```diff
@@ -109,18 +109,43 @@ public function hasOptions()
     }
 
     /**
+     * Returns whether this element conatins elements
+     * @return bool
+     */
+    public function hasElements()
+    {
+        return count($this->elements) > 0;
```

Comment:
Another optimisation, if this is likely to be called frequently. Add ```private $hasElements = false;```; flip this switch either here when counting for the first time, or even better when addElement and skip a potentially costly count altogether.